cpp/CMakeLists.txt

```diff
@@ -76,7 +76,7 @@ if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
 
   option(ARROW_JEMALLOC
     "Build the Arrow jemalloc-based allocator"
-    ON)
+    OFF)
```

Comment:
I can add an external project but I skipped that as it packages nearly everywhere (maybe not on windows but Homebrew and all linux distros come with a decently recent version).